test/unit_tests/braket/circuits/test_circuit.py

```diff
@@ -570,6 +572,184 @@ def test_add_verbatim_box_result_types():
         )
 
 
+def test_measure():
+    circ = Circuit().h(0).cnot(0, 1).measure([0])
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.H(), 0))
+        .add_instruction(Instruction(Gate.CNot(), [0, 1]))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    assert circ == expected
+
+
+def test_measure_int():
+    circ = Circuit().h(0).cnot(0, 1).measure(0)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.H(), 0))
+        .add_instruction(Instruction(Gate.CNot(), [0, 1]))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    assert circ == expected
+
+
+def test_measure_multiple_targets():
+    circ = Circuit().h(0).cnot(0, 1).cnot(1, 2).cnot(2, 3).measure([0, 1, 3])
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.H(), 0))
+        .add_instruction(Instruction(Gate.CNot(), [0, 1]))
+        .add_instruction(Instruction(Gate.CNot(), [1, 2]))
+        .add_instruction(Instruction(Gate.CNot(), [2, 3]))
+        .add_instruction(
+            Instruction(Measure(qubit_count=3, ascii_symbols=["M", "M", "M"]), [0, 1, 3])
+        )
+    )
+    assert circ == expected
+    assert circ._measure_targets == [0, 1, 3]
+
+
+def test_measure_with_noise():
+    circ = Circuit().x(0).x(1).bit_flip(0, probability=0.1).measure(0)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(Gate.X(), 0))
+        .add_instruction(Instruction(Gate.X(), 1))
+        .add_instruction(Instruction(BitFlip(probability=0.1), 0))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    assert circ == expected
+
+
+def test_measure_verbatim_box():
+    circ = Circuit().add_verbatim_box(Circuit().x(0).x(1)).measure(0)
+    expected = (
+        Circuit()
+        .add_instruction(Instruction(compiler_directives.StartVerbatimBox()))
+        .add_instruction(Instruction(Gate.X(), 0))
+        .add_instruction(Instruction(Gate.X(), 1))
+        .add_instruction(Instruction(compiler_directives.EndVerbatimBox()))
+        .add_instruction(Instruction(Measure(), 0))
+    )
+    expected_ir = OpenQasmProgram(
+        source="\n".join(
+            [
+                "OPENQASM 3.0;",
+                "bit[1] b;",
+                "qubit[2] q;",
+                "#pragma braket verbatim",
+                "box{",
+                "x q[0];",
+                "x q[1];",
+                "}",
+                "b[0] = measure q[0];",
+            ]
+        ),
+        inputs={},
+    )
+    assert circ == expected
+    assert circ.to_ir("OPENQASM") == expected_ir
+
+
+def test_measure_in_verbatim_subcircuit():
+    message = "Cannot measure a subcircuit inside a verbatim box."
+    with pytest.raises(ValueError, match=message):
+        Circuit().add_verbatim_box(Circuit().x(0).x(1).measure(0))
```

Comment:
Is there a technical reason we can't allow measurement inside a verbatim box? As long as that qubit isn't touched again for the remainder of the circuit, it seems like it should be ok - I imagine this is normally what a user would try to do first.

Reply:
Currently, a measurement inside the verbatim box ends up being a mid-circuit measurement since all qubits are measured if there is no explicit `measure` added to the circuit. (ie. behind the scenes all circuits would basically be `Circuit.h(0).measure()`)

Measurements inside the verbatim box can definitely be added if you see this as a common use case. We can not add the measurements to the outer circuit if there is a measurement inside the verbatim circuit.

Reply:
Oh, I see - so the measured qubits inside the verbatim box don't currently get added to _measure_targets of the outer circuit?

Are there other similar cases where we would need to detect measurements that might be hidden in some nested subcircuit? Like a subroutine?

(By the way, this is something that could likely be addressed in a later PR.)

Reply:
Correct, the measured qubits inside the verbatim box don't currently get added to the _measure_targets of the outer circuit. 

It is definitely worth testing any similar cases of measurements inside a nested subcircuit. Do you have any cases in mind that I can test? Otherwise, I can try to get a list together to make sure these get tested in a follow-up PR.

Reply:
I don't know of any other ways to nest a subcircuit inside a circuit like this - there may not be any, which would be great.